master/buildbot/util/protocol.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Portions Copyright Buildbot Team Members
+
+
+from twisted.internet import protocol
+
+
+class LineBuffer:
+    def __init__(self, delimiter):
+        self._buffer = b''
+        self._delimiter = delimiter
+
+    def add_data(self, data):
+        # returns lines that have been processed, if any
+        lines = (self._buffer + data).split(self._delimiter)
+        self._buffer = lines.pop(-1)
+        return [l.rstrip(b'\r') for l in lines]
+
+    def get_trailing_lines(self):
```

Comment:
```suggestion
    def get_trailing_line(self):
```
there is only one line..

Reply:
Fixed.